wasm/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+#[wasm_bindgen(js_name = keyPairFromSecretKey)]
+pub fn from_secret_key(secretkey: &[u8]) -> Result<JsValue, JsValue> {
+	// use 64bit for secret key to maintain compatibility with tweetnacl
+	let keypair = Keypair::from_bytes(&secretkey).unwrap();
+	let secret_key = keypair.to_bytes().to_vec();
+	let public_key = keypair.public.as_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	return Ok(JsValue::from_serde(&result).unwrap());
+}
+
+/// Create key pair from secret key for use with ED25519 from ed25519_dalek
+/// * seed: secret key UIntArray with 32 elements
+///
+/// * returned struct has two fields public and secret.
+#[wasm_bindgen(js_name = keyPairFromSeed)]
+pub fn keypair_from_seed(seed: &[u8]) -> Result<JsValue, JsValue> {
+	// use 64bit for secret key to maintain compatibility with tweetnacl
+	let secret = SecretKey::from_bytes(&seed).unwrap();
+	let public: PublicKey = (&secret).into();
+	let keypair = Keypair{secret, public};
+	let public_key = public.as_bytes().to_vec();
+	let secret_key = keypair.to_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	return Ok(JsValue::from_serde(&result).unwrap());
+}
+
+/// Sign a message using ED25519 from ed25519_dalek
+/// * pubkey: UIntArray with 32 elements
+/// * private: UIntArray with 32 or 64 elements
+/// * message: Arbitrary length UIntArray
+///
+/// * returned vector is the signature consisting of 64 bytes.
+#[wasm_bindgen(js_name = signED25519)]
+pub fn ed25519_sign(pubkey: &[u8], seckey: &[u8], message: &[u8]) -> Vec<u8> {
+	let secret: SecretKey = SecretKey::from_bytes(&seckey[..SECRET_KEY_LENGTH]).unwrap();
+	let public: PublicKey = PublicKey::from_bytes(&pubkey[..PUBLIC_KEY_LENGTH]).unwrap();
+	let keypair: Keypair  = Keypair{ secret, public };
+
+    keypair
+		.sign(message)
+		.to_bytes()
+		.to_vec()
+}
+
+/// Verify signature using ED25519 from ed25519_dalek
+/// * pubkey: UIntArray with 32 elements
+/// * signature: UIntArray 64 elements
+/// * message: Arbitrary length UIntArray
+///
+/// * returned vector is the signature consisting of 64 bytes.
+#[wasm_bindgen(js_name = verifyED25519)]
+pub fn ed25519_verify(pubkey: &[u8], signature: &[u8], message: &[u8]) -> bool {
+	let public = PublicKey::from_bytes(&pubkey[..PUBLIC_KEY_LENGTH]).unwrap();	
+		
+	let sig = match Signature::try_from(signature) {
+		Ok(sig) => sig,
+		Err(_) => return false
+	};
+	
+	public
+		.verify(message, &sig)
+		.is_ok()
+}
+
+
+/// Verify signature using ED25519 from ed25519_dalek
+/// * pubkey: UIntArray with 32 elements
+/// * signature: UIntArray 64 elements
+/// * message: Arbitrary length UIntArray
+///
+/// * returned vector is the signature consisting of 64 bytes.
+#[wasm_bindgen(js_name = isOnCurveED25519)]
+pub fn ed25519_is_on_curve(pubkey: &[u8]) -> bool {
+	return curve25519_dalek::edwards::CompressedEdwardsY::from_slice(&pubkey)
+	.decompress()
+	.is_some();
+}
+
+#[wasm_bindgen(js_name = sha256)]
+pub fn hash_sha256(data: &[u8]) -> String {
```

Comment:
Is this intended for derived addresses or did you have something else in mind? I think we would be better off not exporting this since it will likely only be used internally for derived addresses

Reply:
I was pllaning to use this instead of crypto-hash to reduce external dependecies.